src/cpu-kernels/awkward_RegularArray_combinations.cpp

```diff
@@ -0,0 +1,51 @@
+// BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/master/LICENSE
+
+#define FILENAME(line) FILENAME_FOR_EXCEPTIONS_C("src/cpu-kernels/awkward_RegularArray_combinations.cpp", line)
+
+#include "awkward/kernels.h"
+#include "awkward/kernel-utils.h"
+
+template <typename C, typename T>
+ERROR awkward_RegularArray_combinations(
+  T** tocarry,
+  int64_t* toindex,
+  int64_t* fromindex,
+  int64_t n,
+  bool replacement,
+  int64_t size,
+  int64_t length) {
+  for (int64_t j = 0;  j < n;  j++) {
+    toindex[j] = 0;
+  }
+  for (int64_t i = 0;  i < length;  i++) {
+    int64_t start = size*i;
+    int64_t stop = start + size;
+    fromindex[0] = start;
+    awkward_ListArray_combinations_step<T>(
+      tocarry,
+      toindex,
+      fromindex,
+      0,
+      stop,
+      n,
+      replacement);
+  }
+  return success();
+}
```

Comment:
This function is templated by `C` and `C` is never used, not even once. It gets specialized to `int32_t` only, but that's kind of a throw-away. Are there other cases like this?

Reply:
I don't think so.

Reply:
* awkward_ListArray_getitem_next_range_counts defines `T` but doesn't use it.
   * awkward_RegularArray_combinations defines `C` but doesn't use it.

That's it! They've become a lot easier to scan.

These two things should be fixed; I can see this as being part of the refactor.